rewards/utils/emission_utils.py

```diff
@@ -32,6 +33,39 @@ def get_flat_emission_rate(sett: str, chain: str) -> Decimal:
     return Decimal(get_emission_control(chain).proRataEmissionRate(sett).call() / 1e4)
 
 
+@lru_cache
+def get_nft_control(chain: Network) -> ContractFunctions:
+    return make_contract(
+        EMISSIONS_CONTRACTS[chain]["NFTControl"], Abi.NFTControl, chain
+    )
+
+
+@lru_cache
+def get_nft_weights(chain: Network):
+    nft_control = get_nft_control(chain)
+    schedules = list(
+        map(
+            lambda ws: NFTWeightSchedule(ws[0], ws[1], ws[2], ws[3]),
+            nft_control.getNftWeightSchedules().call(),
+        )
+    )
+    weights = {}
+    for weight_schedule in schedules:
+        key = f"{weight_schedule.addr}-{weight_schedule.nft_id}"
+        if key not in weights:
+            weights[key] = weight_schedule
+        else:
+            curr_schedule = weights[key]
```

Comment:
This is unnecessary, the same function in a more readable way and without creating a new variable is:
```py
weights[key] = weight_schedule if weight_schedule.timestamp > weights[key].timestamp else weights[key]
```

Reply:
We also have to handle the case where there is no entry already in the dict, ie, when weights[key] = None